webhooks/virtualmachine/v1alpha1/validation/virtualmachine_validator.go

```diff
@@ -187,18 +196,25 @@ func (v validator) ValidateUpdate(ctx *context.WebhookRequestContext) admission.
 		validationErrs = append(validationErrs, fieldErr.Error())
 	}
 
-	return common.BuildValidationResponse(ctx, validationErrs, nil)
+	return common.BuildValidationResponse(ctx, validationWarnings, validationErrs, nil)
 }
 
-func (v validator) validateMetadata(ctx *context.WebhookRequestContext, vm *vmopv1.VirtualMachine) field.ErrorList {
+func (v validator) validateMetadata(ctx *context.WebhookRequestContext, vm *vmopv1.VirtualMachine) (admission.Warnings, field.ErrorList) {
 	var allErrs field.ErrorList
+	var allWarnings admission.Warnings
 
 	if vm.Spec.VmMetadata == nil {
-		return allErrs
+		return allWarnings, allErrs
 	}
 
 	mdPath := field.NewPath("spec", "vmMetadata")
 
+	// OvfEnv transport is marked as deprecated.  Emit a warnings to indicate that to users.
+	if vm.Spec.VmMetadata.Transport == vmopv1.VirtualMachineMetadataOvfEnvTransport {
+		allWarnings = append(allWarnings,
+			fmt.Sprint(ovfEnvTransportDeprecated))
+	}
```

Comment:
Should we warn `ExtraConfig` here as well since it's deprecated according to our docs at https://vm-operator.readthedocs.io/en/latest/concepts/workloads/guest/#extraconfig?